remotespark/livyclientlib/log.py

```diff
@@ -2,63 +2,26 @@
 # Distributed under the terms of the Modified BSD License.
 
 from __future__ import print_function
-from datetime import datetime
+import logging, logging.config
 
-from .utils import join_paths, get_magics_home_path, get_instance_id, ensure_path_exists, ensure_file_exists
 from .configuration import get_configuration
 from .constants import Constants
 
-
 class Log(object):
-    """Logger."""
+    """Logger for magics. A small wrapper class around the configured logger described in the configuration file"""
+    logging.config.dictConfig(get_configuration(Constants.logging_config))
 
     def __init__(self, caller_name):
         assert caller_name is not None
-
         self._caller_name = caller_name
-
-        self._level = Constants.normal_level
-        self._path_to_serialize = None
-
-        self._read_config()
-
-    @property
-    def level(self):
-        """One of: 'debug', 'normal'"""
-        return self._level
-
-    def _read_config(self):
-        # Create path name
-        magics_home_path = get_magics_home_path()
-        logs_folder_name = "logs"
-        log_file_name = "log_{}.log".format(get_instance_id())
-        self._path_to_serialize = join_paths(magics_home_path, logs_folder_name)
-        ensure_path_exists(self._path_to_serialize)
-        self._path_to_serialize = join_paths(self._path_to_serialize, log_file_name)
-        ensure_file_exists(self._path_to_serialize)
-
-        # Read log level
-        level = get_configuration(Constants.log_level, Constants.debug_level)
-
-        self._level = level
-
+        self.logger = logging.getLogger(Constants.magics_logger_name)
+    
     def debug(self, message):
-        """Prints if in debug mode."""
-        self._handle_message("DEBUG", message, [Constants.debug_level])
+        self.logger.debug(self._transform_log_message(message))
 
     def error(self, message):
-        """Always shows."""
-        self._handle_message("ERROR", message, [Constants.normal_level, Constants.debug_level])
-
-    def _handle_message(self, level, message, accepted_modes):
-        if self.level in accepted_modes:
-            message = "{}\t{}\t{}\t{}\n".format(str(datetime.utcnow()), level, self._caller_name, message)
+        self.logger.error(self._transform_log_message(message))
 
-            if get_configuration(Constants.log_to_disk, True):
-                self._write_to_disk(message)
-            else:
-                print(message)
+    def _transform_log_message(self, message):
+        return '{}\t{}'.format(self._caller_name, message)
```

Comment:
Unit testing is tough here because there's a tight coupling between the contents of the configuration file and the behavior of the logger. If I modify the contents of the configuration file, how will that interfere with the other tests? Keep in mind that the get_configuration function caches the file contents.